onnxruntime/contrib_ops/cpu/transformers/generation_device_helper.h

```diff
@@ -64,11 +64,13 @@ using CreateGptInputsFunc = std::function<Status(
     OrtValue& expanded_attention_mask)>;
 
 using AddToFeedsFunc = std::function<Status(
-    const IExecutionProvider* provider,
     Stream* ort_stream,
     std::initializer_list<OrtValue> inputs,
     std::vector<OrtValue>& feeds,
-    IAllocatorUniquePtr<char>& buffer)>;
+    IAllocatorUniquePtr<char>& buffer,
+    AllocatorPtr gpu_allocator,
+    AllocatorPtr pinned_allocator,
+    const OrtMemoryInfo& location)>;
```

Comment:
should we just provide the allocator map, and let the implementation to decide which allocator to use? 
put gpu_allocator / pinned_allocator in arguments doesn't looks like a flexible design.

Reply:
In this function implementation (GenerationCudaDeviceHelper::AddToFeeds), there is no EP instance, so if we pass in allocator map, we cannot find the key which relies on EP's GetOrtDeviceByMemType() function. So we have to rely on the caller to resolve the desired allocator before calling this function

Reply:
discussed offline, since there is no device_id object in the implementation of AddToFeeds(), we cannot get the corresponding allocator from allocation map. Change the parameter to a more general name "device_allocator" and "host_allocator"